x-pack/plugins/code/index.ts

```diff
@@ -0,0 +1,191 @@
+    require: ['kibana', 'elasticsearch', 'xpack_main'],
+    id: 'code',
+    configPrefix: 'xpack.code',
+    publicDir: resolve(__dirname, 'public'),
+
+    uiExports: {
+      app: {
+        title: 'Code',
+        description: 'Code Search Plugin',
+        main: 'plugins/code/app',
+      },
+      styleSheetPaths: resolve(__dirname, 'public/styles.scss'),
+    },
+
+    config(Joi: any) {
+      return Joi.object({
+        enabled: Joi.boolean().default(true),
+        queueIndex: Joi.string().default('.code-worker-queue'),
+        // 1 hour by default.
+        queueTimeout: Joi.number().default(moment.duration(1, 'hour').asMilliseconds()),
+        // The frequency which update scheduler executes. 5 minutes by default.
+        updateFrequencyMs: Joi.number().default(moment.duration(5, 'minute').asMilliseconds()),
+        // The frequency which index scheduler executes. 1 day by default.
+        indexFrequencyMs: Joi.number().default(moment.duration(1, 'day').asMilliseconds()),
+        // The frequency which each repo tries to update. 1 hour by default.
+        updateRepoFrequencyMs: Joi.number().default(moment.duration(1, 'hour').asMilliseconds()),
+        // The frequency which each repo tries to index. 1 day by default.
+        indexRepoFrequencyMs: Joi.number().default(moment.duration(1, 'day').asMilliseconds()),
+        // timeout a request over 30s.
+        lspRequestTimeoutMs: Joi.number().default(moment.duration(10, 'second').asMilliseconds()),
+        repos: Joi.array().default([]),
+        maxWorkspace: Joi.number().default(5), // max workspace folder for each language server
+        isAdmin: Joi.boolean().default(true), // If we show the admin buttons
+        disableScheduler: Joi.boolean().default(true), // Temp option to disable all schedulers.
+        enableGlobalReference: Joi.boolean().default(false), // Global reference as optional feature for now
+        redirectToNode: Joi.string(),
+      }).default();
+    },
+
+    init: async (server: Server, options: any) => {
+      const queueIndex: string = server.config().get('xpack.code.queueIndex');
+      const queueTimeout: number = server.config().get('xpack.code.queueTimeout');
+      const adminCluster = server.plugins.elasticsearch.getCluster('admin');
+      const dataCluster = server.plugins.elasticsearch.getCluster('data');
+      const log = new Log(server);
+      const serverOptions = new ServerOptions(options, server.config());
+
+      if (serverOptions.redirectToNode) {
+        log.info(
+          `redirect node enabled,all requests will be redirected to ${serverOptions.redirectToNode}`
+        );
+        redirectRoute(server, options, log);
+        await redirectSocketRoute(server, options, log);
+        return;
+      }
+
+      const socketService = new SocketService(server, log);
+
+      // Initialize search clients
+      // @ts-ignore
+      const repoSearchClient = new RepositorySearchClient(dataCluster.getClient(), log);
```

Comment:
Would you mind adding the `getClient()` method to the [`Cluster` class](https://github.com/elastic/kibana/blob/66b3e48c6c9ba07e9378cecb0692536f8879ca80/src/legacy/core_plugins/elasticsearch/index.d.ts#L150-L154) instead of just ignoring this and the next three lines?

Reply:
we could, but getClient returns ESClient class defined in code,

we are definitely going to fix those @ts-ignore by not using them actually
https://github.com/elastic/code/issues/738